### DISCUSSION TITLE
Request to add the board M5Stack CORE2 to tinyuf2/ports/espressif/boards

### DISCUSSION THREAD

**Discussion by PaulskPt** [adafruit/tinyuf2#258] [Q&A] [answered]
The [M5Stack_CORE2](https://shop.m5stack.com/products/m5stack-core2-esp32-iot-development-kit) has a dual core ```Espressif ESP32-D0WDQ6-V3``` processor. It has: 240MHz dual core, 600 DMIPS, 520KB SRAM, Wi-Fi, Bluetooth, 16MB Flash and 8MB PSRAM.

The M5Stack_CORE2 is not listed in circuitpython.org/downloads,
however the following M5Stack boards are:

Result of search: https://circuitpython.org/downloads?q=m5stack :

m5stack_stamp_c3 by Unknown
m5stack_core_fire by Unknown
m5stack_core_basic by Unknown 
m5stack_atom_lite by Unknown 

The 'm5stack_core_basic is a predecessor of the m5stack_core2

These boards are not listed in https://github.com/adafruit/tinyuf2/ports/espressif/boards

I would like to create a firmware.bin / firmware.uf2 for the M5Stack CORE2.

Thank you.

**Comment by hathach** [comment]
esp32 (and also C3) does not have native usb controller like s2 or s3 and therefore it is not possible to support these chips.

**Comment by PaulskPt** [comment]
OK, thank you hathach!

**Comment by PaulskPt** [comment]
Afterburner: 
@hathach. Can you explain me please. I am not very wel 'at home' with the expression: 'native usb controller'.
What I see is that the product page of the M5Stack CORE2 tells this. See the image:

![m5stack_core2_USB_drive](https://user-images.githubusercontent.com/9107950/199047879-878f5b90-3d6f-4bad-9749-37c398c4e1f0.png)

Then, the device is mentioned in the MS Windows 11 Pro Device manager as soon as I connect the device to the desktop PC using an USB-A to USB-C cable. See image:

![2022-10-31_15h30utc_M5Stack_Core2_in_MS_Windows11_Pro_Device_Manager](https://user-images.githubusercontent.com/9107950/199048190-eaad6ac8-faee-4ca1-8b0c-e13872a742b9.png)

**Comment by dhalbert** [comment, marked as answer]
The ESP32 and ESP32-C3 do not have native USB on the ESP chips themselves. They use an external serial-to-USB converter chip. That chip only supports serial, not mass storage. So it is not possible to provide the fake BOOT drive that tinyuf2 provides.

**Comment by PaulskPt** [comment]
Thank you for your explanation Dan. I google'd for that 'native USB' just before my question for more explanation. I did not find what I was looking for. Btw. yesterday I tried to update an NTP sync'ed clock script I wrote earlier this year for the M5Stack CORE2. In the Arduino IDE I received a similar error as you cured for the ESP32-S3-BOX.
